### PR TITLE
elasticsearch: customize user-agent for es client

### PR DESCRIPTION
**Pull request by TattdCodeMonkey**
Updated the ElasticVectorSearch constructor to use a customized user-agent for the elasticsearch client for telemetry.

### CODE REVIEW

File: langchain/src/vectorstores/tests/elasticsearch.int.test.ts [hunk 1]
```diff
@@ -32,7 +31,7 @@ describe("ElasticVectorSearch", () => {
     const indexName = "test_index";
 
     const embeddings = new OpenAIEmbeddings();
-    const store = new ElasticVectorSearch(embeddings, { client, indexName });
+    store = new ElasticVectorSearch(embeddings, { client, indexName });
```
[the remark this answers is not in the document]
Comment (TattdCodeMonkey) [reply langchain/src/vectorstores/tests/elasticsearch.int.test.ts:34]: yes, on line 9 the is `let store: ElasticVectorSearch;` and the test cases are trying to use that. With this being `const store` its creating a new instance and not setting the local variable so the tests `store === undefined` when you actually run them.

[the remark this answers is not in the document]
Comment (TattdCodeMonkey) [reply langchain/src/vectorstores/tests/elasticsearch.int.test.ts:34]: alternatively we could rename this variable `const esStore = new ElasticVectorSearch()` then explicitly set `store = esStore` at the end of the `beforeEach`

**Comment by TattdCodeMonkey** [comment]
I’ll fix the formatting error.

the require i could move, still not sure of the best way to read the package version at runtime.

**Comment by jacoblee93** [comment]
Could we leave it out for now maybe? Or try using import at the top level?

**Comment by TattdCodeMonkey** [comment]
@jacoblee93 I tried a few different things but went ahead a removed the runtime version. Getting the current version would likely best be done using a script to write it to a constant, like in [this answer](https://stackoverflow.com/a/67701490/2434916). And that seems like a bit much for what I'm doing here.

**Comment by jacoblee93** [comment]
Yeah let's leave it out for now. Thank you!

**Pull request merged**